src/main.cpp

```diff
@@ -600,9 +593,14 @@ int main(int argc, char** argv) {
             std::string sourceFilename = baseFilename + ".cpp";
 
             bool withSharedLibrary;
-            std::ofstream os(sourceFilename);
-            synthesiser->generateCode(os, baseIdentifier, withSharedLibrary);
-            os.close();
+            const bool emitToStdOut =
+                    Global::config().has("generate") && Global::config().get("generate") == "-";
+            if (emitToStdOut)
+                synthesiser->generateCode(std::cout, baseIdentifier, withSharedLibrary);
+            else {
+                std::ofstream os{baseFilename + ".cpp"};
```

Comment:
The equivalent `sourceFilename` already exists so use that.
```suggestion
                std::ofstream os{sourceFilename};
```